### PR TITLE
feat: add multiply method to Calculator class

### LINKED ISSUE #9: Add a method to multiply two numbers
@claude Add a method to multiply two numbers

### PR DESCRIPTION
**Pull request by yangxiaohu660**
Add multiplication functionality to the Calculator class to support multiplying two numbers.

## Changes
- Add `multiply` method that takes two integers and returns their product
- Update main method to demonstrate multiplication functionality
- Add comprehensive tests covering basic multiplication, zero cases, and negative numbers

Fixes #9

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/main/java/com/example/Calculator.java [hunk 1]
```diff
@@ -25,6 +29,9 @@ public static void main(String[] args) {
         int subtractResult = calculator.subtract(10, 4);
         System.out.println("10 - 4 = " + subtractResult);
         
+        int multiplyResult = calculator.multiply(4, 6);
```
Comment (yangxiaohu660) [inline src/main/java/com/example/Calculator.java]: @claude 変数multiplyResultをmultipliedResultにリネームしてください。
